tools/shared.py

```diff
@@ -2511,7 +2511,7 @@ def minify_wasm_js(js_file, wasm_file, expensive_optimizations, minify_whitespac
       js_file = Building.acorn_optimizer(js_file, passes)
     # if we can optimize this js+wasm combination under the assumption no one else
     # will see the internals, do so
-    if not Settings.LINKABLE:
```

Comment:
Isn't LINKABLE set in bot those cases anyway?

Reply:
Not in main and side module mode 2 - in that mode, we do still do dce, as if the code were not linkable. This allows better code sizes, but requires a lot more care in picking the exports.